### PR TITLE
Makes lighting objects objects again

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

FOR TOOO LONG WE HAVE SUFF-
Ok yeah so like, we made them overlays to save on maptick, but with threaded maptick that is potentially not an issue anymore.
I'm opening this pr so I can tm it and see. If it works, it'll save about 50% of lighting object update costs, which is pretty damn good.

I'm also removing the fullbright light icon state, since it is barely ever used (it was added as a clientside op, but we don't hit fullbright very much at all so it does nothing but eat my cpu time)

Also also changing how SSlighting does its resolution. Rather then waiting for all sources to process, then working on corners and objects, instead we will do all the sources we had at the start, then all the sources after, and so on.
The goal is to avoid churn causing the system to constantly choke. it is better to potentially double operate then it is for things to feel horrifically slow.

## Why It's Good For The Game

Faster. Also you won't be able to see lights through walls anymore, so mesons will be less dumb. Can maybe bump their nightvision a bit idk we'll see.

## Changelog
:cl:
fix: You can no longer see lights through walls when using mesons or when ventcrawling
/:cl:

### CODE REVIEW

File: code/modules/lighting/lighting_corner.dm [hunk 1]
```diff
@@ -209,7 +205,7 @@
 	else
 		display.pixel_y = -16
 
-	display.color = rgb(cache_r * 255, cache_g * 255, cache_b * 255)
+	display.color = display.color = rgb(cache_r * 255, cache_g * 255, cache_b * 255)
```
Comment (Kylerace) [inline code/modules/lighting/lighting_corner.dm:208]: ?

Comment (LemonInTheDark) [reply code/modules/lighting/lighting_corner.dm:208]: it wasn't me officer

**Comment by san7890** [comment]
Undraft when CI is passing.

**Comment by LemonInTheDark** [comment]
Needs tm don't merge

**Comment by Lorwp** [comment]
Seems to be causing the cargo shuttle to not be able to launch. Reproduction:

1. Try to move the cargo shuttle
2. Get told off for your hubris
3. Remove `/atom/movable/lighting_object ` from `GLOB.blacklisted_cargo_types`
4. Find the shuttle is now happy

cc @Higgin

**Comment by Potato-Masher** [comment]
I suspect this issue is related to this testmerge:

If a tile isn't very well lit, you cannot examine it at all (except for the turfs directly next to your character). This is most noticeable in rooms when the station's night lighting is active making everything dimmer. 

This took me a while to figure out.; I only noticed this in a dimly lit maint room adjacent to space. Space could be examined including things out there but nothing in the room unless it was next to my character.

**Comment by lessthnthree** [comment]
Tram lighting is also weird with it.

image

**Comment by MrMelbert** [comment]
I can confirm that this causes `examine` checks to break due to turf luminosty and dynamic luminosity being 0.

**Comment by Cyberboss** [comment]
Does this mean we can revert the plane cube changes that lock out `vis_contents` shuttles?

**Comment by LemonInTheDark** [comment]
no, sadkirby

**Comment by optimumtact** [comment]
Dont fucking RED label things without a fucking good reason, if you're not working on it, the PR doesn't need to be open.

**Pull request closed without merging**